src/tests/conduit/t_conduit_node_convert.cpp

```diff
@@ -55,84 +55,145 @@
+    CONDUIT_NATIVE_UNSIGNED_CHAR_ID, CONDUIT_NATIVE_UNSIGNED_SHORT_ID, CONDUIT_NATIVE_UNSIGNED_INT_ID, CONDUIT_NATIVE_UNSIGNED_LONG_ID,
+    CONDUIT_NATIVE_FLOAT_ID, CONDUIT_NATIVE_DOUBLE_ID
+};
+
+const NodeConvertFun CONVERT_TO_FUNS[20] = {
+    convert_to_int8_array, convert_to_int16_array, convert_to_int32_array, convert_to_int64_array,
+    convert_to_uint8_array, convert_to_uint16_array, convert_to_uint32_array, convert_to_uint64_array,
+    convert_to_float32_array, convert_to_float64_array,
+    convert_to_char_array, convert_to_short_array, convert_to_int_array, convert_to_long_array,
+    convert_to_unsigned_char_array, convert_to_unsigned_short_array, convert_to_unsigned_int_array, convert_to_unsigned_long_array,
+    convert_to_float_array, convert_to_double_array
+};
+
+const NodeConvertFun CONVERT_ZCIP_FUNS[20] = {
+    convert_zcip_int8_array, convert_zcip_int16_array, convert_zcip_int32_array, convert_zcip_int64_array,
+    convert_zcip_uint8_array, convert_zcip_uint16_array, convert_zcip_uint32_array, convert_zcip_uint64_array,
+    convert_zcip_float32_array, convert_zcip_float64_array,
+    convert_zcip_char_array, convert_zcip_short_array, convert_zcip_int_array, convert_zcip_long_array,
+    convert_zcip_unsigned_char_array, convert_zcip_unsigned_short_array, convert_zcip_unsigned_int_array, convert_zcip_unsigned_long_array,
+    convert_zcip_float_array, convert_zcip_double_array
+};
+
 //-----------------------------------------------------------------------------
 TEST(conduit_node_convert, to_arrays)
 {
-    uint8 data_vals[] = {1,2,3,4,5,6,7,8};
-    
-    Node n;
-    n.set(data_vals,8);
-
-    n.schema().print();
-    
-    Node nconv;
-    
-    // signed bw-style
-    n.to_int8_array(nconv);
-    nconv.print();
-
-    n.to_int16_array(nconv);
-    nconv.print();
-
-    n.to_int32_array(nconv);
-    nconv.print();
-    
-    n.to_int64_array(nconv);
-    nconv.print();
-
-    // unsigned bw-style
-    n.to_uint8_array(nconv);
-    nconv.print();
-
-    n.to_uint16_array(nconv);
-    nconv.print();
-
-    n.to_uint32_array(nconv);
-    nconv.print();
-    
-    n.to_uint64_array(nconv);
-    nconv.print();
-    
-    // float bw-style
-    n.to_float32_array(nconv);
-    nconv.print();
-    
-    n.to_float64_array(nconv);
-    nconv.print();
-
-    // signed native c
-    n.to_char_array(nconv);
-    nconv.print();
-
-    n.to_short_array(nconv);
-    nconv.print();
-
-    n.to_int_array(nconv);
-    nconv.print();
-    
-    n.to_long_array(nconv);
-    nconv.print();
-
-    // unsigned native c
-    n.to_unsigned_char_array(nconv);
-    nconv.print();
-
-    n.to_unsigned_short_array(nconv);
-    nconv.print();
-
-    n.to_unsigned_int_array(nconv);
-    nconv.print();
-    
-    n.to_unsigned_long_array(nconv);
-    nconv.print();
-
-    // float native c
-    n.to_float_array(nconv);
-    nconv.print();
-    
-    n.to_double_array(nconv);
-    nconv.print();
-
-
```

Comment:
I actually prefer the old way here.  Its much more to type, but its a lot clearer what is being tested

Reply:
Also The new form also doesn't save us that much typing b/c of all the c-style functions you had to provide ...

Reply:
I'll concede that the old version was more clear, but it was also only testing the bare minimum in terms of functionality (i.e. that the `to_[type]_array` functions all ran without throwing errors for simple data) and extending it to perform any additional validation tests would make it much longer than the new code.